jetbrains-core/src/software/aws/toolkits/jetbrains/ui/clouddebug/StartupCommandWithAutofile.kt

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.ui.clouddebug
+
+import com.intellij.execution.configurations.RuntimeConfigurationError
+import com.intellij.ide.DataManager
+import com.intellij.openapi.project.Project
+import com.intellij.ui.components.fields.ExpandableTextField
+import com.intellij.ui.components.labels.LinkLabel
+import com.intellij.util.ui.JBUI
+import net.miginfocom.swing.MigLayout
+import software.aws.toolkits.jetbrains.services.clouddebug.CloudDebuggingPlatform
+import software.aws.toolkits.jetbrains.services.clouddebug.DebuggerSupport
+import software.aws.toolkits.jetbrains.services.ecs.execution.ArtifactMapping
+import software.aws.toolkits.resources.message
+import javax.swing.JPanel
+
+/**
+ * Component to display startup command with ability to generate command automatically based on existing Artifacts mapping table.
+ *
+ * @param project - [com.intellij.openapi.project.Project] instance
+ * @param containerName - AWS Docker container name
+ */
+class StartupCommandWithAutofile(private val project: Project, private val containerName: String) :
+    JPanel(MigLayout("novisualpadding, ins 0, gap 0, fillx, wrap 2, hidemode 3", "[][min!]")) {
+
+    private val startupCommandField = ExpandableTextField()
+    private val generateAutomaticallyLink =
+        LinkLabel(message("cloud_debug.run_configuration.generate_automatically.link_text"), null, ::onGenerateAutomaticallyLinkClicked)
+
+    var command: String
+        get() = startupCommandField.text
+        set(value) { startupCommandField.text = value }
+
+    var generateAutomaticallyPopupContent: () -> List<ArtifactMapping> = { emptyList() }
```

Comment:
Rename generateAutomaticPopupContent?

Reply:
Renamed button to "Auto-fill...". I have some concerns about naming for the button. I would like to make it clear for users from one point and keep it short enough from the other hand. That was the main reason I decided not to go with any sort of image for "Auto-fill". Please feel free for any suggestions here.